metals/src/main/scala/scala/meta/internal/builds/BuildTool.scala

```diff
@@ -30,6 +31,15 @@ abstract class BuildTool {
 
   def executableName: String
 
+  def gitignore(workspace: AbsolutePath, paths: List[String]): Unit = {
+    val gitignore = workspace.resolve(".gitignore")
+    if (gitignore.exists) {
+      paths
+        .filterNot(gitignore.readText.contains)
```

Comment:
We're re-reading the `.gitignore` file on every path. It would be better to bind the value of `readText` to a variable.

Reply:
@olafurpg makes sense! I will do that